rt3/tools/python/lib/generate_token_java.py

```diff
@@ -0,0 +1,172 @@
+#! /usr/bin/env python3
+# This script generates token related files from Grammar/Tokens:
+
+from pathlib import Path
+from generate_token import NT_OFFSET, load_tokens, update_file, main
+
+
+# Modified from generate_token to better match Java style indents
+def generate_chars_to_token(mapping, nesting=1, level=1):
+    result = []
+    write = result.append
+    tab = "    "
+    indent = tab * nesting
+    write(tab)
+    write(indent)
+    write("switch (c%d) {\n" % (level,))
+    for c in sorted(mapping):
+        write(indent)
+        value = mapping[c]
+        if isinstance(value, dict):
+            write(tab * 2)
+            write("case '%s':\n" % (c,))
+            write(generate_chars_to_token(value, nesting + 2, level + 1))
+            write(indent)
+            write(tab * 3)
+            write("break;\n")
+        else:
+            write(tab * 2)
+            write("case '%s': return %s;\n" % (c, value))
+    write(tab)
+    write(indent)
+    write("}\n")
+    return "".join(result)
+
+
+token_java_template = """\
+/* Auto-generated by src/python/generate_token_java.py */
+
+package uk.co.farowl.vsj3.evo1.parser;
```

Comment:
Prefer to make this an argument: one less thing to remember when porting to Jython 3.